spec/controllers/report_controller_spec.rb

```diff
@@ -1223,6 +1223,33 @@
       EvmSpecHelper.local_miq_server
     end
 
+    context "when generating reports" do
+      render_views
+      let(:rpt) { FactoryGirl.create(:miq_report) }
+
+      before :each do
+        stub_user(:features => :all)
+
+        seed_session_trees('report', :reports_tree, "xx-0_xx-0-1_rep-#{controller.to_cid(rpt.id)}")
+        session[:sandboxes]["report"][:rep_tree_build_time] = rpt.created_on
+        session[:sandboxes]["report"][:active_accord] = :reports
+      end
+
+      it "runs report and calls GTL generation" do
+        expect_any_instance_of(GtlHelper).to receive(:render_gtl).with match_gtl_options(
+          :model_name                     => 'MiqReportResult',
+          :report_data_additional_options => {
+            :named_scope => [[:with_current_user_groups_and_report, rpt.id]],
+            :model       => 'MiqReportResult'
+          }
+        )
+
+        post :x_button, params: { :pressed => 'miq_report_run', :id => rpt.id }
```

Comment:
Please don't mix the two hash definition styles, use the rockets as miq-bot asks for it:
```ruby
post :x_button, :params => { :pressed => 'miq_report_run', :id => rpt.id }
```
![evilest](https://user-images.githubusercontent.com/649130/33598492-44f461fe-d9a3-11e7-8b1d-d6ad80317536.gif)